src/bun.js/node/util/parse_args.zig

```diff
@@ -0,0 +1,684 @@
+                if (value != null) index += 1;
+            },
+
+            .long_option_and_value => {
+                // e.g. --foo=barconst
+                const equal_index = arg.indexOfCharU8('=');
+                const long_option = arg.substringWithLen(2, equal_index.?);
+                const value = arg.substring(equal_index.? + 1);
+
+                try tokens.append(Token{ .option = .{
+                    .index = index,
+                    .value = value.toJSConst(globalThis),
+                    .inline_value = true,
+                    .name = long_option.toJSConst(globalThis),
+                    .parse_type = .long_option_and_value,
+                    .raw = arg_jsvalue,
+                    .option_idx = findOptionByLongName(long_option, options),
+                } });
+            },
+
+            .positional => {
+                try tokens.append(Token{ .positional = .{ .index = index, .value = arg.toJSConst(globalThis) } });
+            },
+        }
+    }
+}
+
+/// Create the parseArgs result "tokens" field
+/// This field is opt-in, and people usually don't ask for it,
+/// so only create the js values if they are needed
+pub fn createOutputTokensArray(globalThis: *JSGlobalObject, tokens: []const Token) !JSValue {
+    const kinds_count = @typeInfo(TokenKind).Enum.fields.len;
+    var kinds_jsvalues: [kinds_count]?JSValue = [_]?JSValue{null} ** kinds_count;
+
+    var result = JSC.JSValue.createEmptyArray(globalThis, tokens.len);
+    for (tokens, 0..) |token_generic, i| {
+        const obj_fields_count: usize = switch (token_generic) {
+            .option => |token| if (token.value.isUndefined()) 4 else 6,
+            .positional => 3,
+            .@"option-terminator" => 2,
+        };
+
+        // reuse JSValue for the kind names: "positional", "option", "option-terminator"
+        var kind_idx = @intFromEnum(token_generic);
+        var kind_jsvalue = kinds_jsvalues[kind_idx] orelse kindval: {
+            var val = String.static(@as(string, @tagName(token_generic))).toJSConst(globalThis);
+            kinds_jsvalues[kind_idx] = val;
+            break :kindval val;
+        };
+
+        var obj = JSValue.createEmptyObject(globalThis, obj_fields_count);
+        obj.put(globalThis, ZigString.static("kind"), kind_jsvalue);
+        switch (token_generic) {
+            .option => |token| {
+                obj.put(globalThis, ZigString.static("index"), JSValue.jsNumberFromInt32(token.index));
+                obj.put(globalThis, ZigString.static("name"), token.name);
+                obj.put(globalThis, ZigString.static("rawName"), try token.makeRawNameJSValue(globalThis));
+
+                // only for boolean options, it is "undefined"
+                obj.put(globalThis, ZigString.static("value"), token.value);
+                obj.put(globalThis, ZigString.static("inlineValue"), if (token.value.isUndefined()) JSValue.undefined else JSValue.jsBoolean(token.inline_value));
+            },
```

Comment:
```suggestion
                if (!token.value.isUndefined()) {
```
was something like this meant to be added here?